pandas/__init__.py

```diff
@@ -202,8 +202,29 @@
         FutureWarning,
         stacklevel=2,
     )
-# Don't allow users to use pandas.os or pandas.warnings
-del os, warnings
+# Don't allow users to use pandas.os
+del os
+
+# DeprecationWarning for missing pyarrow
+from pandas.compat.pyarrow import pa_version_under10p1
+
+if pa_version_under10p1:
+    import warnings
+    from pandas.compat._optional import VERSIONS
+
+    warnings.warn(
+        f"""
+Pyarrow will become a future required dependency of pandas,
```

Comment:
```suggestion
Pyarrow will become a future required dependency of pandas in the next major release,
```

I would also add a short reason why it's becoming a required dependency

Reply:
Added a short description talking about more performant data types and better interop.